source/CommonJobs/CommonJobs.MVC.UI/Infrastructure/OAuthHelpers.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Web;
+using Newtonsoft.Json.Linq;
+
+namespace CommonJobs.Mvc.UI.Infrastructure
+{
+    public static class OAuthHelpers
+    {
+        public static string BuildUri(string url, string path, NameValueCollection query)
+        {
+            var uriBuilder = new UriBuilder(url)
+            {
+                Path = path,
+                Query = ConstructQueryString(query)
+            };
+            return uriBuilder.ToString();
+        }
+
+        public static String ConstructQueryString(NameValueCollection parameters, string separator = "&")
```

Comment:
(Not important) Does it make sense to have the separator as a parameter? Will it ever be different than `&`?